server/src/main/java/org/opensearch/common/blobstore/fs/FsBlobContainer.java

```diff
@@ -174,10 +174,15 @@ public InputStream readBlob(String name) throws IOException {
     @Override
     public InputStream readBlob(String blobName, long position, long length) throws IOException {
         final SeekableByteChannel channel = Files.newByteChannel(path.resolve(blobName));
-        if (position > 0L) {
-            channel.position(position);
+        try {
+            if (position > 0L) {
+                channel.position(position);
+            }
+            assert channel.position() == position;
+        } catch (Exception | AssertionError e) {
+            channel.close();
+            throw e;
```

Comment:
Note: It doesn't work for CodeGuru review. 🙁

Reply:
If closing the SeekableByteChannel in this method, there will be a test failure:
```
org.opensearch.common.blobstore.fs.FsBlobContainerTests.testReadBlobRangeCorrectlySkipBytes

REPRODUCE WITH: ./gradlew ':server:test' --tests "org.opensearch.common.blobstore.fs.FsBlobContainerTests.testReadBlobRangeCorrectlySkipBytes" -Dtests.seed=D0F13B3C60FBBE7C -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1" -Dtests.locale=he -Dtests.timezone=Africa/Abidjan -Druntime.java=15

org.opensearch.common.blobstore.fs.FsBlobContainerTests > testReadBlobRangeCorrectlySkipBytes FAILED
    java.nio.channels.ClosedChannelException
...
```

Reply:
let's not take this

Reply:
Thanks for your idea, I have reverted the change in `FsBlobContainer.java`.